### PR TITLE
Fetch node instances for scaling up node groups in parallel.

### PR DESCRIPTION
**Pull request by krzysztof-jastrzebski**

### CODE REVIEW

File: cluster-autoscaler/clusterstate/utils/node_instances_cache.go [hunk 1]
```diff
@@ -92,8 +92,25 @@ func (cache *CloudProviderNodeInstancesCache) removeEntriesForNonExistingNodeGro
...
+
+	// Fetch missing node instances.
+	var wg sync.WaitGroup
+	for _, nodeGroup := range nodeGroups {
+		nodeGroup := nodeGroup
+		if _, found := cache.getCacheEntryLocked(nodeGroup); !found {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				cache.fetchCloudProviderNodeInstancesForNodeGroup(nodeGroup)
```
Comment (losipiuk) [inline cluster-autoscaler/clusterstate/utils/node_instances_cache.go]: Please add error logging

Comment (krzysztof-jastrzebski) [reply cluster-autoscaler/clusterstate/utils/node_instances_cache.go]: Done

Comment (tghartland) [reply cluster-autoscaler/clusterstate/utils/node_instances_cache.go]: The line you added for this always logs even when there is no error
```
node_instances_cache.go:106] Failed to fetch cloud provider node instances for default-worker, error <nil>
```
@krzysztof-jastrzebski @losipiuk

Comment (krzysztof-jastrzebski) [reply cluster-autoscaler/clusterstate/utils/node_instances_cache.go]: Fix in https://github.com/kubernetes/autoscaler/pull/2137

**Pull request merged**